test/test.js

```diff
@@ -644,6 +644,103 @@ suite('GaiaFastList >', function() {
           }));
         });
     });
+
+    test('getItemImageSrc() can return a Blob', function() {
+      el = createList();
+      el.configure({
+        getItemImageSrc(item, i) {
+          return getBlobFromURL(`/base/test/lib/artwork-${i % 10}.jpg`);
+        }
+      });
+
+      el.model = createModel();
+
+      var fastList = this.FastList.lastCall.returnValue;
+      return fastList.rendered
+        .then(() => {
+          return Promise.all([].map.call(el.querySelectorAll('img'), (img, index) => {
+            return new Promise(resolve => img.addEventListener('load', e => {
+              assert.include(img.src, 'blob:');
+              resolve();
+            }));
+          }));
+        });
+    });
+
+    test('getItemImageSrc() is only called once per item', function() {
+      el = createList();
+
+      var getItemImageSrc = sinon.spy((item, i) => {
+        return getBlobFromURL(`/base/test/lib/artwork-${i % 10}.jpg`);
+      });
+
+      el.configure({ getItemImageSrc: getItemImageSrc });
+      el.model = createModel();
+
+      var fastList = this.FastList.lastCall.returnValue;
+      return fastList.rendered
+        .then(() => {
+          return new Promise(resolve => {
+            var firstImage = el.querySelector('img');
+            firstImage.addEventListener('load', function f() {
+              firstImage.removeEventListener('load', f);
+              resolve();
+            });
+          });
+        })
+
+        .then(() => {
+
+          // scroll down
+          el.scrollTop = 400 * 4;
```

Comment:
at another point in this test file we "// wait one tick for re-render", do we need it there?